internal/services/sources_service.go

```diff
@@ -16,7 +16,12 @@ func ListSources(w http.ResponseWriter, r *http.Request) {
 		renderError(w, r, payloads.NewClientInitializationError(r.Context(), "sources client", err))
 		return
 	}
-	resp, err := client.ListSourcesWithResponse(r.Context(), &sources.ListSourcesParams{}, AddIdentityHeader)
+
+	name := r.URL.Query().Get("provider")
+	filter := sources.QueryFilter("[source_type][name]=" + name)
+
+	resp, err := client.ListSourcesWithResponse(r.Context(), &sources.ListSourcesParams{Filter: &filter}, AddIdentityHeader)
```

Comment:
from what I understood this is the correct way to filter their requests but unfortunately it does not work dor me.
When I am requesting sources directly It works: 
```golang
curl --location -g  --request GET 'http://localhost:8000/api/sources/v3.1/sources?filter[source_type][name]=amazon' --header 'x-rh-identity: IDENTITY' 
```

But when I am trying to use our service this is the query that is being formed using: https://github.com/RHEnVision/provisioning-backend/blob/eeb276f93a8629d114b2a89e72ea8c978e6bc3bc/internal/clients/sources/client.gen.go#L1229  
```golang
http://localhost:8000/api/sources/v3.1/sources?filter=%5Bsource_type%5D%5Bname%5D%3Damazon
```
This is not working and I am getting this error from sources side: 
```golang
"message":"[PANIC RECOVER] runtime error: index out of range [1] with length 1 goroutine 121 
```
@lzap , @ezr-ondrej  Can you take a look as well?

Reply:
I found the bug and asked sources on their channel.

Reply:
Okay thanks for info, ping us once this is ready for testing. I have my local setup already.